File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++){
            int b = target - nums[i];
            if(map.containsKey(b)){
                int[] ans = {i, map.get(b)};
                return ans;
            }
            else map.put(nums[i], i);
        }
        return new int[2];
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        nmap = {}
        for i in range(len(nums)):
            rem = target - nums[i]
            if rem in nmap.keys():
                return [nmap[rem], i]
            nmap[nums[i]] = i
        return []
